src/substrait/textplan/parser/SubstraitPlanRelationVisitor.cpp

```diff
@@ -435,7 +521,25 @@ std::any SubstraitPlanRelationVisitor::visitExpressionConstant(
 
 std::any SubstraitPlanRelationVisitor::visitExpressionColumn(
     SubstraitPlanParser::ExpressionColumnContext* ctx) {
+  auto relationData =
+      ANY_CAST(std::shared_ptr<RelationData>, currentRelationScope_->blob);
+
+  std::string symbolName = ctx->getText();
+  auto currentFieldNumber = std::find_if(
+      relationData->fieldReferences.begin(),
+      relationData->fieldReferences.end(),
+      [&](auto ref) { return (ref->name == symbolName); });
+
   ::substrait::proto::Expression expr;
+  if (currentFieldNumber != relationData->fieldReferences.end()) {
+    expr.mutable_selection()
+        ->mutable_direct_reference()
+        ->mutable_struct_field()
+        ->set_field(static_cast<int32_t>(
+            (currentFieldNumber - relationData->fieldReferences.begin()) &
+            0xffff));
```

Comment:
Why `& 0xFFFF`?  Should we assert that this value is less than 2 bytes?

Reply:
Changed to use std::numeric_limits<int32_t>::max